src/gui/qgscolorrampbutton.h

```diff
@@ -0,0 +1,323 @@
+/***************************************************************************
+    qgscolorrampbutton.h - Color ramp button
+     --------------------------------------
+    Date                 : November 27, 2016
+    Copyright            : (C) 2016 by Mathieu Pellerin
+    Email                : nirvn dot asia at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#ifndef QGSCOLORRAMPBUTTON_H
+#define QGSCOLORRAMPBUTTON_H
+
+#include "qgscolorrampbutton.h"
+
+#include "qgscolorramp.h"
+#include "qgsstyle.h"
+
+#include <QToolButton>
+
+class QgsPanelWidget;
+
+/** \ingroup gui
+ * \class QgsColorRampButton
+ * A cross platform button subclass for selecting color ramps. Will open color ramp dialogs when clicked.
+ * Offers live updates to button from color ramp dialog. An attached drop down menu allows for access to
+ * saved color ramps, as well as option to invert the current color ramp and create new ramps.
+ * \note Added in version 3.0
+ */
+
+class GUI_EXPORT QgsColorRampButton : public QToolButton
+{
+    Q_OBJECT
+    Q_ENUMS( Behaviour )
+    Q_PROPERTY( QString colorRampDialogTitle READ colorRampDialogTitle WRITE setColorRampDialogTitle )
+    Q_PROPERTY( bool acceptLiveUpdates READ acceptLiveUpdates WRITE setAcceptLiveUpdates )
+    Q_PROPERTY( bool showMenu READ showMenu WRITE setShowMenu )
+    Q_PROPERTY( Behaviour behaviour READ behaviour WRITE setBehaviour )
+    Q_PROPERTY( QgsColorRamp* defaultColorRamp READ defaultColorRamp WRITE setDefaultColorRamp )
+    Q_PROPERTY( QString context READ context WRITE setContext )
+
+  public:
+
+    /** Specifies the behaviour when the button is clicked
+     */
+    enum Behaviour
+    {
+      ShowDialog = 0, //!< Show a color ramp dialog when clicked
+      SignalOnly //!< Emit colorRampClicked signal only, no dialog
+    };
+
+    /** Construct a new color ramp button.
+     * @param parent The parent QWidget for the dialog
+     * @param cdt The title to show in the color chooser dialog
```

Comment:
`color chooser` should be `color ramp`, right? Can you rename this to `const QString& dialogTitle = QString()` and avoid the acronym? (see https://github.com/qgis/qgis3.0_api/issues/53)